lib/Infer/ExhaustiveSynthesis.cpp

```diff
@@ -434,6 +438,20 @@ APInt getNextInputVal(Inst *Var,
   llvm::report_fatal_error("Model does not contain the guess input variable");
 }
 
+Inst *findConst(souper::Inst *I, std::string ConstName) {
```

Comment:
I think this function has exponential runtime, for an expression that has much branching and merging. we've seen this many times and it's something that we need to try hard to avoid. the solution (see many places in souper) is to keep a map of visited instructions.

Reply:
That would be true if it was accumulating all consts.
This one returns the first one it finds (as we only support synthesizing single constants for now).